broker/hosts.py

```diff
@@ -6,30 +6,45 @@
 
 
 class Host:
-
     default_timeout = 0  # timeout in ms, 0 is infinite
 
-    def __init__(self, hostname=None, name=None, from_dict=False, **kwargs):
-        # Allow the class to construct itself from kwargs
-        if from_dict:
-            self.__dict__.update(kwargs)
-        else:
-            self.hostname = hostname or kwargs.get("ip", None)
-            if not self.hostname:
-                # check to see if we're being reconstructued, likely for checkin
-                import inspect
-                if any(f.function == "reconstruct_host" for f in inspect.stack()):
-                    logger.debug("Ignoring missing hostname and ip for checkin reconstruction.")
-                else:
-                    raise HostError("Host must be constructed with a hostname or ip")
-            self.name = name
-        self.username = kwargs.get("username", settings.HOST_USERNAME)
-        self.password = kwargs.get("password", settings.HOST_PASSWORD)
-        self.timeout = kwargs.get(
+    def __init__(self, **kwargs):
+        """Create a Host instance
+
+        :param kwargs: Keyword Arguments:
+            * *hostname* (``str``) --
+            Hostname or IP address of the host, required
+            * *name* (``str``) --
+            Name of the host
+            * *username* (``str``) --
+            Username to use for SSH connection
+            * *password* (``str``) --
+            Password to use for SSH connection
+            * *connection_timeout* (``int``) --
+            Timeout for SSH connection
+            * *port* (``int``) --
+            Port to use for SSH connection
+            * *key_filename* (``str``) --
+            Path to SSH key file to use for SSH connection
```

Comment:
I think this is a little bit extra 😃 
Maybe we just do something like this.
```suggestion
        Expected kwargs:
          hostname: str - Hostname or IP address of the host, required
          name: str - Name of the host
          username: str - Username to use for SSH connection
          password: str - Password to use for SSH connection
          connection_timeout: int - Timeout for SSH connection
          port: int - Port to use for SSH connection
          key_filename: str - Path to SSH key file to use for SSH connection
```